music_assistant/server/providers/deezer/__init__.py

```diff
@@ -143,6 +143,7 @@ async def handle_setup(self) -> None:
 
         self.gw_client = GWClient(self.mass.http_session, self.config.get_value(CONF_ACCESS_TOKEN))
         await self.gw_client.setup()
+        self.config.name = f"Deezer - {self.client.user.name}"
```

Comment:
this is actually not supposed to be changed this way, you are overriding the user's config.
instead override the name property:

```
@property
    def name(self) -> str:
        """Return (custom) friendly name for this provider instance."""
       if client := getattr(self, 'client', None):
         return f"Deezer - {client.user.name}"
       return super().name
```